db/people.go

```diff
@@ -34,3 +42,52 @@ func GetOrCreateUser(ctx *api.Context, user api.Person) (api.Person, error) {
 	err := ctx.DB().Table("people").Create(&user).Error
 	return user, err
 }
+
+type CreateUserRequest struct {
+	Username   string
+	Password   string
+	Properties models.PersonProperties
+}
+
+func CreatePerson(ctx *api.Context, name, personType string) (*models.Person, error) {
+	person := models.Person{Name: name, Type: personType}
+	if err := ctx.DB().Clauses(clause.Returning{}).Create(&person).Error; err != nil {
+		return nil, err
+	}
+
+	return &person, nil
+}
+
+const (
+	// The draft RFC(https://tools.ietf.org/html/draft-irtf-cfrg-argon2-03#section-9.3) recommends
+	// the following time and memory cost as sensible defaults.
+	timeCost    = 1
+	memoryCost  = 64 * 1024
+	parallelism = 4
+	keyLength   = 20
+	saltLength  = 12
+)
+
+func CreateAccessToken(ctx *api.Context, personID uuid.UUID, name, password string, expiry time.Duration) (string, error) {
+	saltRaw := make([]byte, saltLength)
+	if _, err := crand.Read(saltRaw); err != nil {
+		return "", err
+	}
+	salt := base64.URLEncoding.EncodeToString(saltRaw)
+
+	hash := argon2.IDKey([]byte(password), []byte(salt), timeCost, memoryCost, parallelism, keyLength)
+	encodedHash := base64.URLEncoding.EncodeToString(hash)
+
+	accessToken := &models.AccessToken{
+		Name:      fmt.Sprintf("agent-%d", time.Now().Unix()),
+		Value:     encodedHash,
+		PersonID:  personID,
+		ExpiresAt: time.Now().Add(expiry), // long-lived token
+	}
+	if err := ctx.DB().Create(&accessToken).Error; err != nil {
+		return "", err
+	}
+
+	formattedHash := fmt.Sprintf("%s.%s.%d.%d.%d", password, salt, timeCost, memoryCost, parallelism)
+	return formattedHash, nil
```

Comment:
Should we store hash generation meta in the database ? If we change the algo/logic later on, it would be easier to support older logic

Reply:
If we want to prepare ourselves for any change in algo in future, then I think we can add a suffix that can identify the algo used to generate the token.

@moshloop what do you think?